CONTRIBUTING.md

```diff
@@ -0,0 +1,23 @@
+# Welcome
+
+We are glad you are here to help. This document goes over the process of submitting changes and making improvements.
+
+## Developer Steps
+
+Step 5 is optional. It helps avoid submiting broken PRs.
+
+1. Open an Issue
+2. Create a Branch for your work with issue number at the end
+3. Make your changes. Don't forget to update app version in `Makefile`
+4. Check your changes by running [`make scan-build`](./README.md#clang-analyzer)
+5. To validate push your changed to `main` branch to test. This runs git actions to verify build
```

Comment:
Are you sure of this one?

Reply:
LOL, no I am not sure. I very much would like a contribution guide, and I wrote down the steps I use. Open to changes. Any specific suggestions?

Reply:
I would have just deleted the step 5.
So that you have somehow:
- create a branch and make your changes on it.
- don't forget to bump the version
- run test locally: build, build without security warning and functional tests
- push your changes and create a pull request so that: CI tests will be run and reviewer can review.

To me, no one should push to the "`main` branch to test". Tests can be launch locally and while creating the PR.